src/components/RegisterModal/RegisterModal.js

```diff
@@ -0,0 +1,95 @@
+import React, { useState } from "react";
+import { Alert, Modal } from "react-bootstrap";
+
+const RegisterModal = ({ onCloseModal, handleRegistration, registerErr }) => {
+  const [user, setUser] = useState({ name: "", email: "", password: "" });
+  const [errors, setErrors] = useState({ name: "", email: "", password: "" });
+
+  const handleChange = (e) => {
+    setUser({ ...user, [e.target.name]: e.target.value });
+  };
+
+  const validateForm = () => {
+    let formErrors = {};
+    if (!user.name) formErrors.name = "Name is required";
+    if (!user.email) formErrors.email = "Email is required";
+    if (!user.password) formErrors.password = "Password is required";
+    setErrors(formErrors);
+    return Object.keys(formErrors).length === 0;
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    if (!validateForm()) return;
+    try {
+      handleRegistration(user);
+      onCloseModal();
```

Comment:
`      onCloseModal();
`

You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well